packages/core-components/src/components/AlertDisplay/AlertDisplay.tsx

```diff
@@ -57,6 +79,17 @@ export function AlertDisplay(props: AlertDisplayProps) {
     };
   }, [alertApi]);
 
+  useEffect(() => {
+    const [current] = messages;
+    if (current && current.display === 'transient') {
+      const timer = setTimeout(() => {
+        setMessages(msgs => msgs.filter(msg => msg !== current));
+      }, timeoutMs);
+      return () => clearTimeout(timer);
```

Comment:
Don't think we want to cancel this on update, that'd overwrite existing timeouts right? I believe a mount check in the timeout callback could be a simple workaround?

Reply:
Alright, I'm a little embarrassed to say but I've spent a few hours on this and I'm still not totally sure what you're asking me to do. Honestly I just don't know the React and TS/JS stack as well as I'd like :(

I've tested this in a bunch of different ways and it all seems to be working as expected. I feel like I'm missing something obvious

Reply:
Let me see if I can create a failing test for this 😅

Reply:
Sorry I ended up pushing fixes too, kinda had to validate the test with the expected behavior because it is a bit tricky to be sure it's working as expected 😅

I separated out the `firstMessage` extraction outside of the timeout effect, that way we only run the effect whenever the displayed messages changes. That way you get the desired stacking behavior that handles both timeouts and manual removals.

I had a separate fix first, which was to check whether the component was still mounted in the timeout and skipping the `clearTimeout`. It works fine when you're only dealing with the timeouts, but when you mix it with manual removal as well you end up with duplicated timeouts that mess up the display.

Could you pull down the changes on your end too and make sure it's all good?